pyro/distributions/transforms/householder.py

```diff
@@ -21,14 +21,17 @@ class ConditionedHouseholder(Transform):
     event_dim = 1
     volume_preserving = True
 
-    def __init__(self, u_unnormed=None):
+    def __init__(self):
         super().__init__(cache_size=1)
-        self.u_unnormed = u_unnormed
+
+    def _params(self):
+        raise NotImplementedError()
```

Comment:
This seems a bit awkward, creating an object then setting a private member after construction. What if instead we allow tensor-or-callable values in the constructor, similar to `pyro.param`?
```py
class ConditionedHousholder(Transform):
    def __init__(self, u_unnormed):
        super().__init__(cache_size=1)
        assert isinstance(u_unnormed, torch.Tensor) or callable(u_unnormed)
        self.u_unnormed = u_unnormed

    def u(self):
        u_unnormed = self.u_unnormed
        if callable(u_unnormed):
            u_unnormed = u_unnormed()
        ...
```
and correspondingly updated `ConditionalHouseholder`
```py
class ConditionalHouseholder(...):
    def _u_unnormed(self, context):
        # u_unnormed ~ (count_transforms, input_dim)
        # Hence, input_dim must divide
        u_unnormed = self.nn(context)
        if self.count_transforms == 1:
            u_unnormed = u_unnormed.unsqueeze(-2)
        else:
            u_unnormed = torch.stack(u_unnormed, dim=-2)
        return u_unnormed
        
    def condition(self, context):
        u_unnormed = functools.partial(self._u_unnormed, context)
        return ConditionedHouseholder(u_unnormed)
```

Reply:
Ah, that's a much better way of doing that! I'll make these changes :)

Reply:
Done!